### ISSUE TITLE
Errors for Literal[] type hints (comment version)

### ISSUE THREAD

**Issue by basepi** [microsoft/pylance-release#2213]
## Environment data

-   Language Server version: v2021.12.2
-   OS and version: macOS 11.6.1, VSCode 1.63.2
-   Python version (& distribution if applicable, e.g. Anaconda): Multiple, currently activated venv is 3.8.3

## Expected behaviour

For this code snippet:

```python
from typing import Union, Literal


def test(
    style="%",  # type: Union[Literal["%"], Literal["{"], Literal["$"]]
):
    pass
```

I expect no errors. This type hint works fine as a py3 type hint, but when formed as a comment as above, the errors below occur.

## Actual behaviour

Three errors:

<img width="482" alt="Screen Shot 2022-01-04 at 10 20 24" src="https://user-images.githubusercontent.com/702318/148098267-dcdad8d0-d9dd-4cbc-b5d1-0bfa15d14ff6.png">

## Logs

I doubt logs are necessary in this case but can add them if requested.

**Comment by basepi** [comment]
(We're dropping py2 (finally) in the near future, but figured it still might be worth fixing.)

**Comment by erictraut** [comment]
A couple of thoughts:

1. `Literal` allows you to specify multiple literal values, so there's no need to use a `Union` in this case. It would normally be written `Literal["%", "{", "$"]`.
2. You're using an old-style Python 2 type annotation comment. The `Literal` form was introduced after type annotation syntax was added to Python 3, so that explains why this has never been reported previously.

The `Literal` form requires special logic in the parser because a quoted subexpression in a normal type annotation is meant to be parsed as a type expression, but in literals it cannot be parsed as such. When pylance parses an entire source file, it knows that the symbol `Literal` was imported from `typing`, so it can apply this special-casing. But when pylance is parsing a type comment, it's doing so without the benefit of knowing about the `Literal` import, so the special-casing isn't being applied here. I'll need to think more about how to fix this. We may decide not to fix it, since Python 2 was never a focus for Pylance.

**Comment by erictraut** [comment]
I found a reasonably straightforward solution. The fix will be in the next release.

**Comment by basepi** [comment]
Thanks! Definitely would have understood a "won't fix" but glad it will be fixed anyway! :)

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.1.0, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202210-5-january-2022)
